File: 5/H.go
package main

import "fmt"

func main() {

	var n, k int

	fmt.Scan(&n, &k)

	var s string
	fmt.Scan(&s)

	d := make(map[byte]int)
	max_index, max_len := 0, 0
	i, j := 0, 0

	for j < n {
		for j < n && d[s[j]] < k {
			d[s[j]]++
			j++
		}

		if j - i > max_len {
			max_len = j - i
			max_index = i
		}

		d[s[i]]--
		i++
	}

	fmt.Printf("%d %d\n", max_len, max_index + 1)
}	

File: 5/H.py
from collections import defaultdict

n, k = (int(x) for x in input().split())

s = input()

i, j = 0, 0
max_len = 0
max_index = 0
d = defaultdict(lambda : 0)

while j < n:
	while j < n and d[s[j]] <	 k:
		d[s[j]] += 1
		j += 1
		
	cur_len = j - i
	if cur_len > max_len:
		max_len = cur_len
		max_index = i

	d[s[i]] -= 1
	i += 1

max_index += 1

print(max_len, max_index)
